### PR TITLE
DSND-631: Introduce Request authenticator filter to check Eric header…

### PR DESCRIPTION
**Pull request by sameerinfo2003**
[…s in request (Security)](https://companieshouse.atlassian.net/browse/DSND-631)

Screenshot 2022-05-20 at 07 57 43

### CODE REVIEW

File: src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileControllerITest.java [hunk 1]
```diff
@@ -47,20 +47,52 @@ void getCompanyProfileWithMatchingCompanyNumber() throws Exception {
...
+
+        ResponseEntity<CompanyProfile> companyProfileResponse = restTemplate.exchange(
+                COMPANY_URL, HttpMethod.GET, new HttpEntity<Object>(headers),
+                CompanyProfile.class);
 
         assertThat(companyProfileResponse.getStatusCode()).isEqualTo(HttpStatus.OK);
     }
 
+    @Test
+    @DisplayName("Return 401 Unauthorised response when retrieving a company profile without passing Eric headers")
```
Comment (zahidm3) [inline src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileControllerITest.java:62]: Probably convert this test to BDD style feature file?

File: src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+public class EricTokenAuthenticationFilter extends OncePerRequestFilter {
+
+    private final Logger logger;
+
+    public EricTokenAuthenticationFilter(Logger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java:20]: Please write a unit test for this filter

File: src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        if (ericIdentity == null || ericIdentity.isBlank()) {
+            logger.error("Unauthorised request received without eric identity");
+            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        String ericIdentityType = request.getHeader("ERIC-Identity-Type");
+
+        if (ericIdentityType == null || ericIdentityType.isBlank()
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java:33]: We can probably use Apache StringUtils

File: src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                || !(ericIdentityType.equalsIgnoreCase("key")
+                || (ericIdentityType.equalsIgnoreCase("oauth2")))) {
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java:35]: I dont think this logic in if condition is correct, please check again, maybe write a test and pass "oauth2" as header value and see

File: src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            logger.error("Unauthorised request received without eric identity type");
+            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    @Override
+    protected boolean shouldNotFilter(HttpServletRequest request)
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/profile/auth/EricTokenAuthenticationFilter.java:45]: Do we need this exclusion as we excluded already in application config?

**Review by zahidm3 — CHANGES_REQUESTED** [review]

**Review by zahidm3 — APPROVED** [review]

**Review by amobarak — APPROVED** [review]

**Pull request merged**